src/geom/bounding_box.C

```diff
@@ -133,4 +133,25 @@ Real BoundingBox::signed_distance(const Point & p) const
     }
 }
 
+
+void BoundingBox::scale(const Real factor)
+{
+  auto scale_dim = [&](const unsigned int dim) {
+    if (this->first(dim) != std::numeric_limits<Real>::max())
```

Comment:
This can't be right?  In the degenerate case first(dim) would be *negative* max, no?

We should also test second(dim) here, lest someone have a half-infinite box in that direction.

Reply:
```
  void invalidate ()
  {
    for (unsigned int i=0; i<LIBMESH_DIM; i++)
      {
        this->first(i)  =  std::numeric_limits<Real>::max();
        this->second(i) = -std::numeric_limits<Real>::max();
      }
  }
```

Reply:
Ah, okay. I mistook "invalid" for "(half-)infinite", thought you were trying to handle the latter cases.